### DISCUSSION TITLE
Why isn't --target-colorspace-hint enable by default?

### DISCUSSION THREAD

**Discussion by laichiaheng** [mpv-player/mpv#14512] [Q&A] [answered]
This feature is necessary for HDR videos, I wonder why it hasn't been enabled by default? Is there any bug in it?

**Comment by Andarwinux** [comment]
Most people don't have a good HDR monitor, usually a "HDR400" compatible monitor.

**Reply by laichiaheng** [reply]
MiniLED with DisplayHDR1000 is getting much cheaper these years.

**Reply by llyyr** [reply]
Buy me one then I'll send a PR to make it default

**Reply by Andarwinux** [reply]
Sure, MiniLED are an economical choice if buying a new monitor, but you need to consider the monitors that exist on the market: they don't automatically upgrade to HDR1000, and most people don't replace their monitors until they break. Furthermore, HDR1000 is extremely rare in mobile devices.

**Comment by hooke007** [comment]
Why is it necessary for hdr sources?

**Reply by laichiaheng** [reply]
Without it, you can't play HDR videos in HDR mode.

**Reply by hooke007** [reply]
Then just set it if you want. I don't see the valid reason for `default`

**Reply by laichiaheng** [reply]
It will do unnecessary tone-mapping to SDR, it doesn't make sense.

**Reply by hooke007** [reply]
IMO it's preference. Tonemapping to SDR is always reliable for many devices.

**Reply by laichiaheng** [reply]
Doesn't `--target-colorspace-hint` work for both SDR and HDR mode? If it does, it is more compatible with more devices than without it.

**Reply by kasper93** [reply]
Consumer HDR monitors are bad and built-in tonemapping is even worse. Unless you get top of the line one, but then you are aware what you should enable. Tonemapping in mpv is almost always better choice for those "HDR400" monitors.

It is preference and the ecosystem is not ready for `--target-colorspace-hint` to be default. You have an option, enable it if you want.

**Reply by hooke007** [reply]
> Doesn't --target-colorspace-hint work for both SDR and HDR mode? If it does, it is more compatible with more devices than without it.

Don't ignore @Andarwinux 's comment.

**Reply by kasper93** [reply]
And yes, in perfect world, `--target-colorspace-hint` should be enabled by default or even be always enabled, but there are so many things in-between outside mpv control. That's mostly why it is not default to produce consistent output. Up until recently `--target-colorspace-hint` was not working at all on linux, and it still require specific setup.

**Reply by lauyv** [reply]
> And yes, in perfect world, `--target-colorspace-hint` should be enabled by default or even be always enabled, but there are so many things in-between outside mpv control. That's mostly why it is not default to produce consistent output. Up until recently `--target-colorspace-hint` was not working at all on linux, and it still require specific setup.是的，在完美的世界中， `--target-colorspace-hint`应该默认启用，甚至始终启用，但是 mpv 控制之外还有很多东西。这就是为什么它不是默认产生一致输出的主要原因。直到最近`--target-colorspace-hint`在 Linux 上根本不起作用，并且仍然需要特定的设置。

why now `yes` be setted default value? it's time 'ecosystem is ready for '? just personal confuse, because of commit 23843b4aa594dc8c885575f3d237cde3c29398a2.

**Reply by llyyr** [reply]
You can set it to to `no` in your `mpv.conf` if your monitor can't do HDR well.

**Comment by SuperSamus** [comment, marked as answer]
mpv defaults to `--target-colorspace-hint=auto` since v0.41.0.
